DataFormats/Common/interface/ThinnedRefSet.h

```diff
@@ -0,0 +1,50 @@
+#ifndef DataFormats_Common_ThinnedRefSet_h
+#define DataFormats_Common_ThinnedRefSet_h
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/RefItemGet.h"
+
+#include <unordered_set>
+
+namespace edm {
+  class EDProductGetter;
+
+  template <typename C>
+  class ThinnedRefSet {
+  public:
+    class Filler {
+    public:
+      Filler(ThinnedRefSet<C>* set, RefProd<C> refProd, edm::EDProductGetter const& prodGetter)
+          : set_(set), refProd_(refProd), prodGetter_(prodGetter) {}
+
+      template <typename T, typename F>
+      void insert(Ref<C, T, F> const& ref) {
+        if (ref.isNull())
+          return;
+        auto thinnedRef = thinnedRefFrom(ref, refProd_, prodGetter_);
+        set_->keys_.insert(thinnedRef.key());
```

Comment:
`thinnedRefFrom()` may return a Null `Ref`, so the return value should be checked for that and ignored if it is Null.

Reply:
Done